### PR TITLE
2.0

### PR DESCRIPTION
**Pull request by j4mie**
So.. this PR does quite a lot, and when it is released will be backwards incompatible. This was necessary to support anything newer than Django 2.2.

django-db-queue is now quite old, and relies on a bunch of unmaintained dependencies. This PR removes them all, at the cost of making the upgrade path a bit of a pain. Details are in the README but in short: you have to drop the jobs table and recreate it.

We now use the cross-database `JSONField` implementation that was added in Django 3.1, rather than a third-party unmaintained thing. I've also removed the dependency on my own [simplesignals](https://github.com/j4mie/simplesignals/) library (which I haven't touched in nine years!)

I've removed some stuff that was never used (the `create_job` management command (which [didn't work](https://github.com/dabapps/django-db-queue/issues/38) anyway) and the `JobSerializer`, which I'd forgotten even existed.

### CODE REVIEW

**Review by RealOrangeOne — APPROVED** [review]

File: testsettings.py [hunk 1]
```diff
@@ -1,21 +1,15 @@
 import os
 import dj_database_url
 
+
+DATABASE_URL = os.environ.get("DATABASE_URL", "sqlite:///:memory:")
```
Comment (jordaneb) [inline testsettings.py:5]: one too many `/` here?

Comment (j4mie) [reply testsettings.py:5]: Weirdly this is actually correct. See https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#connect-strings for an explanation.

Comment (jordaneb) [reply testsettings.py:5]: ok cool :)

**Review by jordaneb — APPROVED** [review]

**Comment by j4mie** [comment]
Superceded by #42

**Pull request closed without merging**